modules/smartadserverBidAdapter.js

```diff
@@ -132,18 +132,19 @@ export const spec = {
    * @returns positive integer value of startdelay
    */
   getStartDelayForVideoBidRequest: function(videoMediaType, videoParams) {
-    if (videoParams !== undefined && videoParams.startDelay) {
+    if (videoParams?.startDelay) {
       return videoParams.startDelay;
-    } else if (videoMediaType !== undefined) {
-      if (videoMediaType.startdelay == 0) {
-        return 1;
-      } else if (videoMediaType.startdelay == -1) {
-        return 2;
-      } else if (videoMediaType.startdelay == -2) {
-        return 3;
+    } else if (videoMediaType) {
+      switch (videoMediaType) {
+        case 0:
+          return 1;
+        case -1:
+          return 2;
+        case -2:
+          return 3;
       }
     }
-    return 2;// Default value for all exotic cases set to bid.params.video.startDelay midroll hence 2.
+    return 1; // SADR-5619
```

Comment:
this appears wrong; what do you do if videoMediaType is say, 180? In this case, you appear to ignore the declaration and guess that it is preroll?

Reply:
Good point, @patmmccann, I've updated the implementation. Thx.

Reply:
If the publisher specifies midroll starting at 3 minutes (180), you are calling it preroll still?

Reply:
You are completely right. It's fixed.

Reply:
Hi @patmmccann, could you review it, please? Thx.